.github/workflows/tests.yml

```diff
@@ -5,50 +5,53 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ['3.9', '3.10', '3.11', '3.12', '3.13-dev']
+        python-version: ['3.9', '3.10', '3.11', '3.12', '3.13']  # add 3.14-dev once numpy/skimage have wheels available
         # https://numpy.org/neps/nep-0029-deprecation_policy.html
         numpy-version: ['1.22', 'latest', 'dev']
         exclude:
           - python-version: '3.12'
             numpy-version: '1.22'
-          - python-version: '3.13-dev'
+          - python-version: '3.13'
+            numpy-version: '1.22'
+          - python-version: '3.14-dev'
             numpy-version: '1.22'
       fail-fast: false
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
       - name: Install Dependencies
         run: |
           set -x
           set -e
           python -m pip install -r requirements-dev.txt
+          python -m pip install -U setuptools
           if [[ ${{ matrix.numpy-version }} == 'latest' ]]; then
               python -m pip install --upgrade numpy
           elif [[ ${{ matrix.numpy-version }} == 'dev' ]]; then
               python -m pip install --pre --upgrade --extra-index https://pypi.anaconda.org/scientific-python-nightly-wheels/simple numpy
           else
               python -m pip install --upgrade numpy==${{ matrix.numpy-version }}.*
           fi
-      - name: Test Installation
-        run: |
-          python -m pip install --verbose .
+  
         # COVERAGE_CORE=sysmon does not work with Cython. Since coverage is
         # really slow without it (see
         # https://github.com/nedbat/coveragepy/issues/1665), just omit Cython
         # coverage in 3.12. Hopefully by the time it is our minimal version it
         # will work with Cython, or trace coverage will be faster again.
-      - name: Set CYTHON_COVERAGE=1
-        run: echo "CYTHON_COVERAGE=1" >> $GITHUB_ENV
-        if: matrix.python-version != '3.12' && matrix.python-version != '3.13-dev'
+      - name: Set CYTHON_COVERAGE=1 and install Cython 3.0
+        if: contains(fromJSON('["3.9", "3.10", "3.11"]'), matrix.python-version)
+        run: |
+          echo "CYTHON_COVERAGE=1" >> $GITHUB_ENV
+          # Cython 3.1.0 broke coverage usage, so install 3.0.x (see gh-203)
+          python -m pip install "Cython<3.1.0"
```

Comment:
This is fine, but I thought the other PR was going to require the newer Cython.

Reply:
It isn't, Cython 3.1.0 is only needed for free-threading support, there's no reason to drop 3.0.x for the default CPython interpreters.

Reply:
OK. We're not going to want to be pinned forever though so we definitely need to try to get this fixed upstream. The current situation of coverage failing on random lines makes it next to impossible to enforce full test coverage.

Reply:
Yep, that's a good follow-up indeed. And I know more people are going to be bitten by this now that 3.1.0 is out - IIRC Oscar was digging into coverage as well already.

In the meantime it's nice to have green CI though:)